specutils/io/default_loaders/jwst_reader.py

```diff
@@ -585,6 +587,24 @@ def _jwst_s3d_loader(filename, **kwargs):
 
             wavelength = Quantity(wavelength_array, unit=wavelength_unit)
 
+            # The GWCS is currently broken for some IFUs, here we work around that
+            wcs = None
+            if wavelength.shape[0] != flux.shape[-1]:
+                # Need MJD-OBS for this workaround
+                if 'MJD-OBS' not in hdu.header:
```

Comment:
Might no longer be necessary with https://github.com/astropy/astropy/pull/12598 in astropy now, though I guess does not hurt to get this working with older astropy versions.